pkg/server/proto/server.proto

```diff
@@ -9,6 +9,11 @@ import "google/protobuf/timestamp.proto";
 import "google/rpc/status.proto";
 import "any.proto";
 
+/********************************************************************
+* PERMANENTLY RESERVED FIELD NUMBERS
+* 100000000 - inline keepalives
```

Comment:
Can we actually reserve this number? I know for the job message we do:

https://github.com/hashicorp/waypoint/blob/5dba36dd4f0eb165905e46e0063709c8fdba63ee/pkg/server/proto/server.proto#L1621

But this is scoped to a single message. I don't think there's a way in protobuf to reserve a field number across every single message so I'm not entirely sure what to do with this here. I think the protobuf should be able to know the message type so it shouldn't treat a field in `message Foo { field = XX; }` the same as `message Bar { field = XX; }` right?

Reply:
Agree - I don't think it's possible to reserve it globally.

The problem is that, if someone added a field 100000000 to a streaming message with type `bytes`, we would fail here: https://github.com/hashicorp/waypoint/pull/3621/files#diff-6aec7bf0685492b402d82d82a0388113f3937282a2ebf463f3674c419850e3b5R56

The proto unmarshaller only looks at the field numbers. It will attempt to unmarshal our keepalive message into the real proto, and because they both use field number 100000000, it won't just end up in "unknown fields", it'll really to unmarshal that field. That will fail, because the types don't match, and we'll get a generic unmarshal error out. 

I've looked, and there's no way for us distinguish that error (keepalive unmarshalled into something else) from any kind of unmarshal error. The only way to guard against it is to not use field 100000000 in any streaming message. For simplicity, my verbage reserves it globally.

I'm not too worried - I think it's likely that we'll be able to retire this code before we ever hit field 100000000 😆.

Reply:
Why would RecvMsg fail if there's another message with the same field number? Or is that link linking to the wrong line?